cmd/pyroscope/command/command.go

```diff
@@ -0,0 +1,70 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pyroscope-io/pyroscope/pkg/cli"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/pyroscope-io/pyroscope/pkg/config"
+)
+
+type cmdRunFn func(cmd *cobra.Command, args []string) error
+
+func createCmdRunFn(cfg interface{}, vpr *viper.Viper, fn cmdRunFn) cmdRunFn {
+	return func(cmd *cobra.Command, args []string) error {
+		var err error
+		if err = vpr.BindPFlags(cmd.Flags()); err != nil {
+			return err
+		}
+		if c, ok := cfg.(config.File); ok {
+			if err = loadConfigFile(c.Path(), cmd, vpr, cfg); err != nil {
+				return fmt.Errorf("loading configuration file: %w", err)
+			}
+		}
+		if err = cli.Unmarshal(vpr, cfg); err != nil {
+			return err
+		}
+		if err = fn(cmd, args); err != nil {
+			cmd.SilenceUsage = true
+		}
+		return err
```

Comment:
```suggestion
		return nil
```
should we return an unintialized variable?

Reply:
Not really :) The check above is just to suppress "usage" message if an error is not related to CLI stuff (e.g. invalid flag/missing args, etc.), and should not affect error handling flow: the error itself (a nil, or not - does not matter) will be handled by the caller.

Alternatively, it could be put as follows:
```go
		if err = fn(cmd, args); err != nil {
			cmd.SilenceUsage = true
			return err
		}
		return nil
```

I don't have any preferences on this. If you find this option more appropriate, make a suggestion - I'll accept it :)

Reply:
Oh great, this makes sense.